tests/v2/validation/provisioning/k3s/secret_encryption_test.go

```diff
@@ -0,0 +1,320 @@
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/secrets"
+	"github.com/rancher/rancher/tests/framework/extensions/sshkeys"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+type SecretEncryptionTestSuite struct {
+	suite.Suite
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+	psact                 string
+	advancedOptions       provisioning.AdvancedOptions
+}
+
+const (
+	etcdLabel         = "node-role.kubernetes.io/etcd"
+	machineAnnotation = "cluster.x-k8s.io/machine"
+	externalIP        = "k3s.io/external-ip"
+	secretKey         = "testKey"
+	secretValue       = "testValue"
+	etcdctlInstallCmd = `ETCD_VER=v3.5.0 && GOOGLE_URL=https://storage.googleapis.com/etcd && GITHUB_URL=https://github.com/etcd-io/etcd/releases/download &&
+	DOWNLOAD_URL=${GOOGLE_URL} && rm -f /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz && rm -rf /tmp/etcd-download-test && mkdir -p /tmp/etcd-download-test &&
+	curl -L ${DOWNLOAD_URL}/${ETCD_VER}/etcd-${ETCD_VER}-linux-amd64.tar.gz -o /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz &&
+	tar xzvf /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz -C /tmp/etcd-download-test --strip-components=1 && rm -f /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz &&
+	/tmp/etcd-download-test/etcd --version && /tmp/etcd-download-test/etcdctl version && /tmp/etcd-download-test/etcdutl version &&
+	sudo cp /tmp/etcd-download-test/etcdctl /usr/local/bin/etcdctl && etcdctl version`
+)
+
+func (s *SecretEncryptionTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *SecretEncryptionTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.ns = defaultNamespace
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	s.k3skubernetesVersions = clustersConfig.K3SKubernetesVersions
+
+	s.cnis = clustersConfig.CNIs
+	s.providers = clustersConfig.Providers
+	s.nodesAndRoles = clustersConfig.NodesAndRoles
```

Comment:
If we need to have a set roles 3 etcd, 2 cp and 3 worker nodes why not just set that here? NodeandRoles should be reserved for dynamic tests.

Reply:
Okay, i've made the required change.